### PR TITLE
Parsed anime tv/movie api json for genres and wrote to anime_genres.json

### PR DESCRIPTION
**Pull request by JarrodColwell**
# Description
Wrote a script to parse the genres from the anime_tv_batch.json and compile them into the anime_tv_genres.json file.

### CODE REVIEW

File: src/server/generate/anime-genres.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import fs from 'fs';
+import AnimeGenre from '../classes/animeGenre';
+import GenreObject from '../classes/genreObject';
+
+const getAnimeGenres = async () => {
+  try {
+    const tvData = fs.readFileSync('../static/anime_tv_batch.json');
+    const movieData = fs.readFileSync('../static/anime_movie_batch.json');
```
Comment (TAsif) [inline src/server/generate/anime-genres.ts]: I think we can read both files at once and combine the data

File: src/server/generate/anime-genres.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    const data = tvData.toString().concat(movieData.toString());
+    const genres: AnimeGenre[] = [];
+    let genreCounter = 1;
```
Comment (TAsif) [inline src/server/generate/anime-genres.ts]: we should be using `uuid` for all ids, likely need to add the package

File: src/server/generate/anime-genres.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                break;
+              }
+            }
+            if (add) {
+              genres.push(new AnimeGenre(genreCounter.toString(), name.toLowerCase()));
+              genreCounter += 1;
+            }
+          }
+          i += 1;
+          line = lines[i];
```
Comment (TAsif) [inline src/server/generate/anime-genres.ts]: There is for sure some data structure that gets rid of duplicates that escapes me at the moment. And reformatting this to use UUIDs will be way safer

File: src/server/generate/anime-genres.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        }
+      }
+    }
+    fs.writeFile('../static/anime_genres.json', JSON.stringify(new GenreObject(genres)), (err) => {
+      if (err) console.log(err);
+    });
+  } catch (e:any) {
```
Comment (TAsif) [inline src/server/generate/anime-genres.ts]: From my limited JS/TS knowledge, I think we can avoid these gross try/catch blocks and use neater callback functions instead

File: src/server/generate/anime-genres.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    const genres: AnimeGenre[] = [];
+    let genreCounter = 1;
+    const lines = data.toString().split('\n');
+    for (let i = 0; i < lines.length; i += 1) {
+      let line = lines[i];
+      if (line.includes('genres')) {
+        i += 1;
+        line = lines[i];
+        while (!line.includes(']')) {
+          if (line.includes('"name":')) {
```
Comment (TAsif) [inline src/server/generate/anime-genres.ts]: There should probably be a way to get a `value` directly from a `key` in a JSON object

**Review by TAsif — COMMENTED** [review]
Initial thoughts while I play around with the code

**Review by madhi-naga — APPROVED** [review]
LGTM

**Review by TAsif — APPROVED** [review]

**Pull request merged**